### PR TITLE
Use vcpkg for dependency management on all platforms

### PR DESCRIPTION
**Pull request by JasonMarechal25**
- Change centos7 image management. Image is now antaresrte/xpansion-centos7 and tags follow semver schema
- Migrate depedency management to vcpkg
- Use action to setup cmake as much as possible
- jsoncpp staticly linked to ease deploiment

Comments and explanations:

- _git fetch --unshallow_ is necessary because action checkout is shallow but vcpkg requires a "full" clone.
- tbb is still manage manualy
- jsoncpp is still link dynamicaly for MSVC due to difficulty to manage static/synamique with MSVC

### CODE REVIEW

File: .github/workflows/build_centos7.yml [hunk 1]
```diff
@@ -72,13 +72,13 @@ jobs:
...
-    container: 'antaresrte/rte-antares:centos7-system-deps'
+    container: 'antaresrte/xpansion-centos7'
     strategy:
       matrix:
         xprs: [
           # { value: XPRESS-ON, ref: 8.13a },
-                { value: XPRESS-ON, ref: 9.2.5 },
-          #        { value: XPRESS-OFF }
+          { value: XPRESS-ON, ref: 9.2.5 },
+          #{ value: XPRESS-OFF }
```
Comment (a-zakir) [inline .github/workflows/build_centos7.yml:81]: remove comment

File: .github/workflows/build_oracle8.yml [hunk 2]
```diff
@@ -101,51 +109,44 @@ jobs:
...
-          pip3 install -r requirements-tests.txt
+          pushd vcpkg
+          git fetch --unshallow
+          ./bootstrap-vcpkg.sh
+          popd
+          vcpkg/vcpkg install
 
       - name: Configure
         run: |
           [[ ${{ matrix.xprs.value }} == "XPRESS-ON" ]] && XPRESS_VALUE="ON" || XPRESS_VALUE="OFF"
```
Comment (a-zakir) [inline .github/workflows/build_oracle8.yml:128]: may be delete xpress matrix

File: .github/workflows/doxygen.yml [hunk 3]
```diff
@@ -21,15 +21,15 @@ jobs:
 
       - name: theme
         run: |
-             git clone https://github.com/jothepro/doxygen-awesome-css.git
-             cd doxygen-awesome-css
-             git checkout v2.2.1
-             git apply ../docs/antares-xpansion.patch
-             
+          git clone https://github.com/jothepro/doxygen-awesome-css.git
```
Comment (a-zakir) [inline .github/workflows/doxygen.yml]: please revert  all auto-formatting,

File: src/cpp/helpers/CMakeLists.txt [hunk 4]
```diff
@@ -38,27 +38,21 @@ add_library (helpers STATIC
...
-		jsoncpp_lib
+		${jsonlib}
 		solvers
 		glog::glog
 		gflags::gflags
 		Boost::boost Boost::program_options
 		MINIZIP::minizip-ng
 )
-
-install(FILES
```
Comment (a-zakir) [inline src/cpp/helpers/CMakeLists.txt:61]: i think these lines can not be deleted for windows installation, right?

File: .github/workflows/build_centos7.yml (hunk 1 above)
Comment (JasonMarechal25) [reply .github/workflows/build_centos7.yml:81]: I'm keeping them for simplicity when debugging with ACT. In local I don't have access to token for xpress repos and must use XPRESS-OFF. I could pass the value in command line but I haven't been able to pass a map value to ACT

File: src/cpp/helpers/CMakeLists.txt (hunk 4 above)
Comment (JasonMarechal25) [reply src/cpp/helpers/CMakeLists.txt:61]: Indeed. I forgot a change in Toplevel CMakelist.

File: .github/workflows/build_oracle8.yml (hunk 2 above)
Comment (JasonMarechal25) [reply .github/workflows/build_oracle8.yml:128]: I open a new issue to track this : https://github.com/AntaresSimulatorTeam/antares-xpansion/issues/781

**Review by a-zakir — APPROVED** [review]

**Pull request merged**